src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java

```diff
@@ -89,11 +92,20 @@ public FinancialTransactionCategoryDTO updateFinancialTransactionCategory(
                 = financialTransactionCategoryRepository.findById(id)
                 .orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
                         String.format("Financial transaction category with id: %d not found", id)));
+
+        User user = getUserByUserId(financialTransactionCategoryUpdateDTO.userId());
+
         financialTransactionCategory.setName(financialTransactionCategoryUpdateDTO.name());
         financialTransactionCategory.setType(financialTransactionCategoryUpdateDTO.type());
+        financialTransactionCategory.setUser(user);
         return financialTransactionCategoryModelMapper
                 .mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(financialTransactionCategory);
     }
 
+    private User getUserByUserId(Long userId) {
+        return (userId == null) ? null : userRepository.findById(userId).orElseThrow(() ->
+                new AppRuntimeException(ErrorCode.U005, String.format("User with id: %d doesn't exist.", userId)));
+    }
+
```

Comment:
Why did you use the ternary operator? When userId == null then return null. When we don't have userId - globalExceptionHandler handle constraintException. We use userRepository.find... when given userId doesn't exist

Reply:
Done!